### ISSUE TITLE
Make SetTrie iterable

### ISSUE THREAD

**Issue by ldeluigi** [BBVA/mercury-settrie#6]
Implement `__iter__` on the SeTrie to allow iterating over all the sets contained.

At the current state I'm using this workaround:

```py
s.supersets({})
```

**Comment by sbasaldua** [comment]
Hi again ldeluigi,

This suggestion requires less work to implement. 

It has two parts: 

1. Return an iterator over the ids. (Easy to do, but you may want the sets too.)
2. Return an iterator over the sets. (This requires adding the parent to each node, which would also be necessary if we finally do .remove()).

Maybe the way to implement it in Python is adding two properties that return iterators (of string for ids and set for sets).

```python

st = settrie.SetTrie()

# some code to insert sets

for my_id in st.ids:
	print(my_id)
	
for my_set in st.sets:
	print(my_set)

for my_id, my_set in zip(st.ids, st.sets):
	pass

```

We will discuss it with the team and come back with some plan to make it happen.

Thanks for your suggestions.

**Comment by sbasaldua** [comment]
Hi again,

Good news!

We just published `mercury-settrie 1.4.4` that allows iterating over the whole object, getting a placeholder for each set and over the elements of each set.

You can find out more looking at the code example: https://github.com/BBVA/mercury-settrie?tab=readme-ov-file#usage
